authors/apps/articles/views.py

```diff
@@ -591,3 +593,71 @@ def delete(self, request, **kwargs):
                 raise Forbidden
         except Comment.DoesNotExist:
             raise ItemDoesNotExist
+
+
+class BookmarkAPIView(GenericAPIView):
+    """
+    class for Bookmarking and unbookmarking
+    """
+    permission_classes = (IsAuthenticated,)
```

Comment:
@Amoswachira You should also add allow `readonly` for the purpose of the `GET` request

Reply:
@Bakley This is a protected route only Authenticated users can get their bookmarks

Reply:
Yes. I agree with @Amoswachira. Only authenticated users should get their bookmarks.